hydroflow/src/compiled/pull/symmetric_hash_join.rs

```diff
@@ -1,24 +1,53 @@
 use std::collections::HashMap;
 
 #[derive(Debug)]
-pub struct JoinState<K, V1, V2> {
-    ltab: HashMap<K, Vec<V1>>,
-    rtab: HashMap<K, Vec<V2>>,
-    lbuffer: Option<(K, V1, Vec<V2>)>,
-    rbuffer: Option<(K, V2, Vec<V1>)>,
+pub struct HalfJoinState<K, VBuild, VProbe> {
+    table: HashMap<K, Vec<VBuild>>,
+    buffer: Option<(K, VProbe, usize)>,
 }
-
-impl<K, V1, V2> Default for JoinState<K, V1, V2> {
+impl<K, VBuild, VProbe> Default for HalfJoinState<K, VBuild, VProbe> {
     fn default() -> Self {
         Self {
-            ltab: HashMap::new(),
-            rtab: HashMap::new(),
-            lbuffer: None,
-            rbuffer: None,
+            table: HashMap::new(),
```

Comment:
no more Lattices?

Reply:
wasn't there before, could be re-added but really would be if we split join into multiple smaller operator pieces